kratos/python/add_search_strategies_to_python.cpp

```diff
@@ -24,10 +24,47 @@
 #include "spatial_containers/specialized_spatial_search_factory.h"
 #include "spatial_containers/geometrical_objects_bins.h"
 #include "spatial_containers/spatial_search_result.h"
+#include "spatial_containers/spatial_search_result_container.h"
 
 namespace Kratos::Python
 {
 
+/**
+ * @brief Converts a vector to a Python list using pybind11.
+ * @details This function is generic enough to be moved to a more general place.
+ * @tparam T The type of the vector.
+ * @param results The vector to convert.
+ * @return The converted Python list.
+ */
+template<typename T>
+pybind11::list VectorToPyList(const T& results) {
+    pybind11::list list_results;
+    for (auto& r_result : results) {
+        list_results.append(r_result);
+    }
+    return list_results;
+}
+
+/**
+ * @brief Converts a matrix to a nested Python list using pybind11.
+ * @details This function is generic enough to be moved to a more general place.
+ * @tparam T The type of the matrix.
+ * @param results The matrix to convert.
+ * @return The converted nested Python list.
+ */
+template<typename T>
+pybind11::list MatrixToPyList(const T& results) {
```

Comment:
This seems generic, would it make sense to move it to define_python.h ?

Reply:
Perhaps